### PR TITLE
Add support for Scala 3.0.0-M3

### PR DESCRIPTION
**Pull request by tgodzik**
In order not to deprecate 0.27.0-RC1 and 3.0.0-M1 we needed to use an older munit version for those, as we needed to drop support for them in munit. There was also small needed fix, since typeRef was no longer available, but paramRef does invoke it underneath. 

Another issues that we faced was the fact the mtags no longer compiled on the 0.4.5 dotty plugin version and we needed to update both it and sbt, which caused some tests to fail and I ignored them for now until we can fix it in Bloop.

### CODE REVIEW

File: tests/slow/src/test/scala/tests/sbt/SbtBloopLspSuite.scala [hunk 1]
```diff
@@ -439,7 +439,8 @@ class SbtBloopLspSuite
     } yield ()
   }
 
-  test("definition-deps") {
+  // TODO latest bloop-sbt plugin does not properly export sources
```
Comment (tgodzik) [inline tests/slow/src/test/scala/tests/sbt/SbtBloopLspSuite.scala:442]: I am looking currently at the sbt-bloop issue. I will create anyway an issue to track this.

**Comment by ckipp01** [comment]
> In order not to deprecate 0.27.0-RC1 and 3.0.0-M1 we needed to use an older munit version for those,

Are we also able to just drop 0.27? I've been seeing a lot of other libraries doing that. Or is it needed for the course still?

**Comment by tgodzik** [comment]
> > In order not to deprecate 0.27.0-RC1 and 3.0.0-M1 we needed to use an older munit version for those,
> 
> Are we also able to just drop 0.27? I've been seeing a lot of other libraries doing that. Or is it needed for the course still?

Apparently they might still work on it in January, so I will leave it for now and avoid any confusion.

File: mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala [hunk 2]
```diff
@@ -119,7 +119,7 @@ case class ScalaPresentationCompiler(
       val sourceFile = CompilerInterfaces.toSource(params.uri, params.text)
       driver.run(uri, sourceFile)
 
-      given ctx as Context = driver.currentCtx
+      val ctx = driver.currentCtx
```
Comment (ckipp01) [inline mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala:122]: 🤔 This isn't 100% clear to me. How come this doesn't need to be a `given` anymore?

File: project/build.properties [hunk 3]
```diff
@@ -1 +1 @@
-sbt.version=1.4.2
+sbt.version=1.4.5
```
Comment (ckipp01) [inline project/build.properties:1]: Do we really want to do this with the test issues?

Comment (ckipp01) [reply project/build.properties:1]: Ah... I suppose we have to if we want to bump sbt-dotty.

File: tests/slow/src/test/scala/tests/sbt/SbtBloopLspSuite.scala (hunk 1 above)
Comment (ckipp01) [reply tests/slow/src/test/scala/tests/sbt/SbtBloopLspSuite.scala:442]: Sounds good. I have create this in sbt as well https://github.com/sbt/sbt/issues/6203

File: mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala (hunk 2 above)
Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/ScalaPresentationCompiler.scala:122]: This is a workaround not to use givens, because they changed the syntax and I would have duplicate the whole file. By doing it all explicitely it seems to work fine.

File: project/build.properties (hunk 3 above)
Comment (tgodzik) [reply project/build.properties:1]: Yeah, I need to finally look into the issue here

File: build.sbt [hunk 4]
```diff
@@ -132,8 +132,8 @@ inThisBuild(
     resolvers += Resolver.sonatypeRepo("public"),
     dependencyOverrides += V.guava,
     // faster publishLocal:
-    publishArtifact.in(packageDoc) := sys.env.contains("CI"),
-    publishArtifact.in(packageSrc) := sys.env.contains("CI"),
+    publishArtifact.in(Compile / packageDoc) := sys.env.contains("CI"),
```
Comment (tgodzik) [inline build.sbt:135]: Funily enough, without this change we would try to publish Tests and it would fail currently.

File: mtags/src/main/scala-3/scala/meta/internal/pc/Scala3CompilerWrapper.scala [hunk 5]
```diff
@@ -12,8 +12,8 @@ class Scala3CompilerWrapper(driver: InteractiveDriver)
   override def compiler(): InteractiveDriver = driver
 
   override def resetReporter(): Unit = {
-    given ctx as Context = driver.currentCtx
-    ctx.reporter.removeBufferedMessages
+    val ctx = driver.currentCtx
+    ctx.reporter.removeBufferedMessages(using ctx)
```
Comment (gabro) [inline mtags/src/main/scala-3/scala/meta/internal/pc/Scala3CompilerWrapper.scala:16]: Out of sheer curiosity, was this change needed or is it just a style preference?

Comment (tgodzik) [reply mtags/src/main/scala-3/scala/meta/internal/pc/Scala3CompilerWrapper.scala:16]: It's so that we can cross compile 3.0.0-M2 and 3.0.0-M3 - given syntax changed between those two versions.

Comment (gabro) [reply mtags/src/main/scala-3/scala/meta/internal/pc/Scala3CompilerWrapper.scala:16]: Aha, smart!

**Comment by tgodzik** [comment]
We need to wait for https://github.com/scalacenter/bloop/pull/1432, since -Ysemanticdb no longer works due to a bug reported here https://github.com/lampepfl/dotty/issues/10844 :/

**Comment by romanowski** [comment]
sbt-dotty `1.5.1` has been just released that should fix the problem with publish in `mtags/test`

**Comment by tgodzik** [comment]
> sbt-dotty `1.5.1` has been just released that should fix the problem with publish in `mtags/test`

Cool! Thanks! I need to fix it in Bloop still unfortunately.

File: tests/unit/src/main/scala/tests/QuickBuild.scala [hunk 6]
```diff
@@ -118,6 +118,7 @@ case class QuickBuild(
           s"org.scala-lang:scala-reflect:$scalaVersion"
         )
       }
+    println(scalaDependencies.toList)
```
Comment (ckipp01) [inline tests/unit/src/main/scala/tests/QuickBuild.scala:121]: 😬 forgotten println

**Review by ckipp01 — APPROVED** [review]
Other than the random pritnln, LGTM! Great job hunting down all these issues and getting them either fixed or figuring out workarounds for all of them 🚀

File: tests/unit/src/main/scala/tests/QuickBuild.scala (hunk 6 above)
Comment (tgodzik) [reply tests/unit/src/main/scala/tests/QuickBuild.scala:121]: Ach, damn! Fixing it now!

**Review by gabro — APPROVED** [review]
LGTM!
